### PR TITLE
feat: improve the playback timeout error experience

### PR DESCRIPTION
**Pull request by kchang-brightcove**
## Description
Ticket: https://brightcove.atlassian.net/browse/WP-8

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -255,17 +255,42 @@ const initPlugin = function(player, options) {
...
+      display.fillWith(content);
+
+      display.getChild('closeButton').hide();
+
+      const reloadButton = display.el().querySelector('.vjs-errors-timeout-button-container > button:first-child');
+      const dismissButton = display.el().querySelector('.vjs-errors-timeout-button-container > button:last-child');
+
+      player.on(reloadButton, 'click', function() {
+        player.reset();
+        player.src(player.options_.sources);
```
Comment (misteroneill) [inline src/plugin.js:287]: player.options_.sources` won't be populated in all cases, I don't think. Here's a suggestion:
```suggestion
        const source = player.currentSource();
        
        player.reset();
        player.src(source);
```

**Review by misteroneill — CHANGES_REQUESTED** [review]
One small suggestion!

File: src/plugin.js (hunk 1 above)
Comment (kchang-brightcove) [reply src/plugin.js]: Implemented suggestion.

File: src/plugin.js [hunk 2]
```diff
@@ -255,17 +255,44 @@ const initPlugin = function(player, options) {
...
+
+      player.on(reloadButton, 'click', function() {
+        const source = player.currentSource();
+
+        player.reset();
+        player.src(source);
+      });
+      player.on(dismissButton, 'click', function() {
+        display.close();
+      });
```
Comment (misteroneill) [inline src/plugin.js:293]: Unfortunately, these listeners create a small memory leak. Even when the display is closed and the elements are in the DOM, the internal Video.js listeners do not go away.

There's not a very easy way to find this, but I set a breakpoint in Video.js and exposed the `DomData` object. Then I tracked the number of listeners counted on this object as I created and dismissed the timeout modal multiple times. Each time the number of listeners increased by two. 

I'd suggest adding the following:
```js
display.one('modalclose', () => {
  player.off(reloadButton);
  player.off(dismissButton);
});
```
That should clear these two listeners when these DOM elements disappear when the display is closed.

**Review by misteroneill — CHANGES_REQUESTED** [review]
Looks good and works like a charm. Noticed one small thing that would only really come through experience with Video.js event model, though.

**Pull request merged**